File: frontend-beheer-2024/src/types/organisationDetails.ts
import { FastApiResponse } from '.'
import { Language } from './misc'

export interface OrganisationDetails {
  about: string
  contact_info: string
  organisation_id: number
  id: number
  create_dt: string
  language: Language
}

export interface OrganisationDetailsResponse extends FastApiResponse {
  data: OrganisationDetails
}

export interface UpdateOrganisationDetailsResponse extends FastApiResponse {
  data: OrganisationDetails
}

File: backend/app/schemas/organisation_details.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from .misc import Language
from .algoritme_version import AlgoritmeVersionQuery


class OrganisationDetailsUpdatable(BaseModel):
    about: str | None = None
    contact_info: str | None = None

    model_config = ConfigDict(from_attributes=True, extra="ignore")


class OrganisationDetailsContent(OrganisationDetailsUpdatable):
    name: str


class OrganisationDetailsIn(OrganisationDetailsContent):
    organisation_id: int
    language: Language


class OrganisationDetailsDB(OrganisationDetailsIn):
    id: int
    create_dt: datetime

    show_page: bool
    code: str
    org_id: str
    type: str


class OrganisationDetailsExport(OrganisationDetailsContent):
    language: Language
    code: str
    org_id: str
    type: str
    show_page: bool


class OrganisationDetailsPage(OrganisationDetailsContent):
    show_page: bool
    algoritme_versions: list[AlgoritmeVersionQuery]
